### PR TITLE
Fix Gluten CI

### PR DESCRIPTION
**Pull request by wForget**
### Why are the changes needed?

Fix Gluten CI

### How was this patch tested?

https://github.com/wForget/kyuubi/actions/runs/12271186564

### Was this patch authored or co-authored using generative AI tooling?

No

### CODE REVIEW

File: .github/workflows/gluten.yml [hunk 1]
```diff
@@ -55,20 +49,45 @@ jobs:
...
+          key: ccache-centos7-release-default-${{github.sha}}
+          restore-keys: |
+            ccache-centos7-release-default
+      - name: Build Gluten velox third party
         run: |
           if [[ "${{ steps.gluten-cache.outputs.cache-hit }}" != 'true' ]]; then
-            git clone https://github.com/oap-project/gluten.git
-            cd gluten
-            ./dev/buildbundle-veloxbe.sh
+            git clone https://github.com/apache/incubator-gluten.git
```
Comment (pan3793) [inline .github/workflows/gluten.yml:65]: use `--depth=1` to skip retrieving git history commits to speed up the clone
```suggestion
            git clone https://github.com/apache/incubator-gluten.git --depth=1
```

File: .github/workflows/gluten.yml [hunk 2]
```diff
@@ -55,20 +49,45 @@ jobs:
...
+            cd incubator-gluten
+            docker run -v $GITHUB_WORKSPACE/incubator-gluten:/work -w /work apache/gluten:vcpkg-centos-7 bash -c "
```
Comment (pan3793) [inline .github/workflows/gluten.yml:67]: How hard would it be if we just built it from scratch?

Comment (wForget) [reply .github/workflows/gluten.yml:67]: This docker image includes the gluten/velox dependencies, reducing the time to download and build dependencies.

**Review by pan3793 — APPROVED** [review]

File: .github/workflows/gluten.yml [hunk 3]
```diff
@@ -18,6 +18,11 @@
 name: Gluten CI
 
 on:
+  pull_request:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/gluten.yml'
```
Comment (pan3793) [inline .github/workflows/gluten.yml:25]: let's change it back to restore the nightly scheduling

**Comment by wForget** [comment]
thanks, merged to master

**Pull request closed without merging**